zerver/lib/actions.py

```diff
@@ -1499,7 +1499,11 @@ def do_send_messages(messages_maybe_none: Sequence[Optional[MutableMapping[str,
             event['sender_queue_id'] = message['sender_queue_id']
         send_event(message['realm'], event, users)
 
-        if url_embed_preview_enabled(message['message']) and links_for_embed:
+        if not links_for_embed:
```

Comment:
I don't understand why we need this, @hashirsarwar. In fact, if `links_for_embed` is an empty set, filtering whitelisted URLs from it wouldn't change anything. I think we can get rid of this whole if block, right?